invenio/modules/oauthclient/contrib/orcid.py

```diff
@@ -122,12 +123,38 @@ def account_info(remote, resp):
 
 def account_setup(remote, token):
     """Perform additional setup after user have been logged in."""
+    from invenio.modules.accounts.models import UserEXT
     from invenio.ext.sqlalchemy import db
     from ..handlers import token_session_key
 
+    import json
+    import requests
+
+    orcid = session.get(token_session_key(remote.name) +
+                        "_account_info").get("external_id")
+
     extra_data = {
-        "orcid": session.get(token_session_key(remote.name) +
-                             "_account_info").get("external_id")
+        "orcid": orcid
         }
     token.remote_account.extra_data = extra_data
     db.session.commit()
+
+    # Fill user full name if not already set
+    user = UserEXT.query.filter_by(id=orcid).first().user
+    if user and not any([user.given_names, user.family_name]):
+        # Query ORCID to get the real name
+        request_url = 'http://orcid.org/{}/orcid-bio'.format(orcid)
+
+        headers = {'Accept': 'application/orcid+json'}
+        response = requests.get(request_url, headers=headers)
+        code = response.status_code
+
+        if code == requests.codes.ok:
+            orcid_bio = json.loads(response.content)
```

Comment:
can you add some error checking (rollback) in case of error in `json.loads`, `KeyError` or `db.session.commit`?